### ISSUE TITLE
source activate is broken for zsh 4.3.10

### ISSUE THREAD

**Issue by pavoljuhas** [conda/conda#6683]
I am using Anaconda on an older Linux box with zsh 4.3.10.  As of 9166defcbdc892aa7c466b9511fe74f36e0c7d5a the `source activate` exits my shell:

```zsh
$ source /path/to/anaconda/bin/activate
Error: activate must be sourced. Run 'source activate envname'
instead of 'activate envname'.
# shell exits
```
This happens because zsh 4.3.10 does not have the `ZSH_EVAL_CONTEXT` parameter yet and the [_conda_script_is_sourced](https://github.com/conda/conda/blob/9d759d8edeb86569c25f6eb82053f09581013a2a/conda/shell/bin/activate#L39-L53) then returns an incorrect result.

@kalefranz - can you please fix that function so it works with zsh 4.3.10?

If it is critical to enforce sourcing of the `activate` script, I feel it would be much simpler to just remove its executable permission.  It would be then impossible to run it otherwise than by sourcing and you could go away with all the complicated tests in `_conda_script_is_sourced`.

**Comment by kalefranz** [comment]
Probably a duplicate of #6639, but with more detail isolating root cause.

**Comment by kalefranz** [comment]
Any reference to a currently-supported distro that has that version of zsh?  Would like to test solutions inside a docker container.

**Comment by pavoljuhas** [comment]
I see this in Ubuntu 10.04.4 which is not supported anymore, but has an ISOs online.

Installing an older zsh from its [git repository](https://github.com/zsh-users/zsh.git) should be quite doable as well.

**Comment by kalefranz** [comment]
@pavoljuhas Can you confirm your still having a problem with the latest conda release?

**Comment by mbargull** [comment]
> If it is critical to enforce sourcing of the `activate` script, I feel it would be much simpler to just remove its executable permission.  It would be then impossible to run it otherwise than by sourcing and you could go away with all the complicated tests in `_conda_script_is_sourced`.

This is exactly what was done as part of https://github.com/conda/conda/pull/6669.
Hence the problematic functionality is completely gone, i.e., issue resolved with `conda >=4.4.11` :tada:.

**Comment by pavoljuhas** [comment]
@kalefranz - conda 4.4.11 is not available yet so I cannot check.  The problem happens with 4.4.10.

**Comment by mbargull** [comment]
If you want to try out the new version before it gets released on the main channel, you can run
```bash
conda update -c conda-canary conda
```

**Comment by pavoljuhas** [comment]
I can confirm the problem is fixed in 4.4.11.  Thank you.
